actions/class.Runner.php

```diff
@@ -57,6 +65,34 @@ public function __construct()
     }
 
     /**
+     * Gets the anti-CSRF manager
+     * @return CsrfToken
+     */
+    protected function getCsrf()
+    {
+        if (!$this->csrf) {
+            $this->csrf = new SessionCsrfToken('TEST_RUNNER');
+        }
+        return $this->csrf;
+    }
+
+    /**
+     * @param $data
+     * @param int $httpStatus
+     */
+    protected function returnJson($data, $httpStatus = 200)
+    {
+        // auto append the CSRF token to the result
+        if (is_array($data)) {
+            $data['token'] = $this->getCsrf()->getCsrfToken();
+        } else if (is_object($data)) {
+            $data->token = $this->getCsrf()->getCsrfToken();
+        }
+
+        parent::returnJson($data, $httpStatus);
```

Comment:
Could you just add a `return` before `parent::returnJson($data, $httpStatus);` as a good practice because the parent method send headers and write out content so we don't want something to append after.